src/test/java/io/strimzi/kafka/bridge/mqtt/MqttBridgetIT.java

```diff
@@ -0,0 +1,238 @@
+    private static final String KAFKA_TOPIC = "devices_bluetooth_data";
+    private static KafkaConsumer<String, String> kafkaConsumerClient = null;
+    private static MqttServer mqttBridge;
+    private static StrimziKafkaContainer kafkaContainer;
+    private String mqttClientId;
+
+
+    /**
+     * Start the kafka Cluster
+     * Start the MQTT bridge before all tests
+     */
+    @BeforeAll
+    public static void beforeAll() throws InterruptedException {
+        String kafkaBootstrapServers = null;
+        try {
+            kafkaContainer = new StrimziKafkaContainer();
+            kafkaContainer.waitForRunning();
+            kafkaContainer.start();
+            kafkaBootstrapServers = kafkaContainer.getBootstrapServers();
+        } catch (Exception e) {
+            logger.error("Exception occurred", e);
+            throw e;
+        }
+
+        // instantiate the kafka consumer client
+        kafkaConsumerClient = new KafkaConsumer<>(
+                ImmutableMap.of(
+                        ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers,
+                        ConsumerConfig.GROUP_ID_CONFIG, "gid-" + UUID.randomUUID(),
+                        ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OffsetResetStrategy.EARLIEST.name().toLowerCase(Locale.ROOT)
+                ),
+                new StringDeserializer(),
+                new StringDeserializer()
+        );
+
+        // consumer client subscribes to the kafka topic
+        kafkaConsumerClient.subscribe(Collections.singletonList(KAFKA_TOPIC));
+
+        // prepare the configuration for the bridge
+        BridgeConfig bridgeConfig = BridgeConfig.fromMap(
+                ImmutableMap.of(
+                        BridgeConfig.BRIDGE_ID, "my-bridge",
+                        KafkaConfig.KAFKA_CONFIG_PREFIX + ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers,
+                        MqttConfig.MQTT_HOST, MQTT_SERVER_HOST,
+                        MqttConfig.MQTT_PORT, MQTT_SERVER_PORT
+                ));
+
+        // prepare the mapping rules
+        String mappingRulesPath = Objects.requireNonNull(MqttBridgetIT.class.getClassLoader().getResource("mapping-rules-regex.json")).getPath();
+        MappingRulesLoader.getInstance().init(mappingRulesPath);
+
+        // start the MQTT bridge
+        CountDownLatch latch = new CountDownLatch(1);
+
+        EventLoopGroup bossGroup = new NioEventLoopGroup();
+        EventLoopGroup workerGroup = new NioEventLoopGroup();
+
+        mqttBridge = new MqttServer(bridgeConfig, bossGroup, workerGroup, ChannelOption.SO_KEEPALIVE);
+        mqttBridge.start();
+
+        latch.countDown();
+        latch.await();
+    }
+
+    /**
+     * Randomly generate a new client id before each test
+     */
+    @BeforeEach
+    public void generateClientId() {
+        mqttClientId = "mqtt-client-" + new Random().nextInt(20);
+    }
```

Comment:
still thinking you don't need this to be a `@BeforeEach`. You can make this private and then calling this method each time you need an MQTT client id in each test. So you don't need the private field to store it.

Reply:
Is there any drawback? 

For me it's good using @BeforeEach to avoid the need of calling generateClientId(); all the time.

Reply:
So something like this ...

```java
private String generateClientId() {
     return "mqtt-client-" + new Random().nextInt(20);
}
```

remove the `mqttClientId` from the class.

Reply:
Looks better.

Reply:
> Is there any drawback? For me it's good using https://github.com/beforeeach to avoid the need of calling generateClientId(); all the time.

It's anyway called all the time. And you are storing the result in a field member which you don't really need which can also drives to side effects :-)

Reply:
I would call it `generateAndGetClientId()` or simpler `getRandomClientId()`

Reply:
yes, got you!